rd-net/Lifetimes/Lifetimes/LifetimedList.cs

```diff
@@ -36,19 +44,31 @@ public Enumerator(LifetimedList<T> list) : this()
       public bool MoveNext()
```

Comment:
That's not quite right. 
lifetime can be terminated after MoveNextб but Current.Value will return the correct value (I mean that value will not be partially cleared in ClearValueIfNotAlive. This may be important for large structures that cannot be read/written atomically)

Reply:
Alright, could you please then formulate the right invariant yourself? It is quite hard to read this method while trying to also figure out which invariant is it supposed to hold.